src/chrome/logging/gdpr.ts

```diff
@@ -0,0 +1,56 @@
+/*---------------------------------------------------------
+ * Copyright (C) Microsoft Corporation. All rights reserved.
+ *--------------------------------------------------------*/
+
+/** Used to store data that can potentially be customer content. If it is, we should never log it or send it over telemetry */
+ export interface PossiblyCustomerContent<T> {
+    customerContentData: T;
+
+    /** Transform the data. If it is customer data, the result will sill be customer data */
+    transform<U>(transformFunction: (customerContentData: T) => U): PossiblyCustomerContent<U>;
+}
+
+/** Used to protect CustomerContent data so we don't inadvertently log it or send it as telemetry */
+export class CustomerContent<T> implements PossiblyCustomerContent<T> {
+    public constructor(data: T) {
+        Object.defineProperty(this, 'customerContentData', {
+            get: () => data
+        });
+    }
+
+    public static maybeNull<T>(data: T | null): PossiblyCustomerContent<T> | null {
+        return data !== null
+            ? new CustomerContent(data)
+            : null;
+    }
+
+    public static maybeUndefined<T>(data: T | undefined): PossiblyCustomerContent<T> | undefined {
```

Comment:
both `maybeNull` and `maybeUndefined` would be more reusable as higher order functions e.g.:

```ts
function mapOrUndefined<T, R>(mappingFunc: (item: T) => R): (item: T) => R | undefined {
    return (item: T) => (item !== undefined) ? mappingFunc(item) : undefined;
}
```

which you could then call like,

```ts
    const myVar = mapOrUndefined(x => new CustomerContent(x))(myString);
```

or, if you wanted to make the convenience function specific as well,

```ts
    export const customerContentOrUndefined( content: string ) = mapOrUndefined(x => new CustomerContent(x);
```

Reply:
I didn't realize this. We are not using those 2 any more. Deleted them.